### ISSUE TITLE
Access private claims via User

### ISSUE THREAD

**Issue by schmitzhermes** [ahus1/keycloak-dropwizard-integration#12]
Hey all,

any chance to access private claims via the `User` object I can inject into my resource method via `@Auth`? I only see the chance to see roles (which is fine), but I may need more information - the JWT is self-contained after all. 
Is there a chance to access those claims? If not, is there any interest in having a PR for that? 

Thanks in advance,
Jan

**Comment by ahus1** [comment]
The idea at design time was that you could either 

(a) use User "as is" as long as it fits your need

(b) provide User with any other class that fits your need (let's call it AlternativeUser)

The steps for (b) would be as follows: provide your own AlternativeUser (extends AbstractUser) together with an AlternativeUserAuthenticator (extends Authorizer<AlternativeUser>) and AlternativeKeycloakAuthenticator (extends AbstractKeycloakAuthenticator<AlternativeUser>) -- instead of the minimal abstract classes you might also be able to extend the default implementations.

When choosing (b) you'll need to override three more methods when calling addBundle (see TODO below). 

```
        bootstrap.addBundle(new KeycloakBundle<LotteryConfiguration>() {
            @Override
            protected KeycloakConfiguration getKeycloakConfiguration(LotteryConfiguration configuration) {
                return configuration.getKeycloakConfiguration();
            }
           /* TODO: override getUserClass(), createAuthorizer() and createAuthenticator() */
        });

```

To summarize: this should give you the possibility to write your own replacement of User as needed _without changing the library_. 

Please let me know if you find this useful, over-engineered or not helpful at all.

If you find that your extension has matured and that an extension of the default User implementation is useful for others, please provide a pull request.

**Comment by schmitzhermes** [comment]
Hey @ahus1,
that's useful advice, thank you very much. I am gonna go ahead and try it. 

> If you find that your extension has matured and that an extension of the default User implementation is useful for others, please provide a pull request.
Thanks - I think I am going to do that. However, I do think that your approach is far more flexible and the PR should only be "on top" and no replacement for your current mechanism.

**Comment by ahus1** [comment]
Ok, let me know how it works out for you. I'm closing this ticket for now - please open this or a new ticket if you have more questions.
